### ISSUE TITLE
Coin/ammo slots and crafting/reforge UI disappears with inactive master mode slot existing

### ISSUE THREAD

**Issue by direwolf420** [bluemagic123/tModLoader#3218]
### Version

Latest 1.4-stable

### OS

Windows

### Platform

Steam

### This bug affects

Gameplay as a Player

### Description

If a player has an item in the extra accessory slot when playing on a master mode world, restarts the game, and then joins another non-master mode world, that slot becomes "inactive". As long as the item is in the slot, the ammo, coin, crafting, and reforge UI disappears until the item is taken out from the slot, which makes the slot itself disappear.

Additionally, switching to the "Equipment" view makes the UI appear again as long as that view is active, except now it has the same "darkened" appearance as the inactive master mode slot.

### Log File

```cs
[13:54:04.002] [Main Thread/WARN] [tML]: Silently Caught Exception: 
System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at Terraria.ModLoader.AccessorySlotLoader.Draw(Int32 skip, Boolean modded, Int32 slot, Color color) in tModLoader\Terraria\ModLoader\AccessorySlotLoader.cs:line 161
   at Terraria.ModLoader.AccessorySlotLoader.DrawAccSlots(Int32 num20) in tModLoader\Terraria\ModLoader\AccessorySlotLoader.cs:line 62
   at Terraria.Main.DrawInventory() in tModLoader\Terraria\Main.cs:line 31562
   at Terraria.Main.DrawInterface_27_Inventory() in tModLoader\Terraria\Main.cs:line 35029
   at Terraria.Main.<SetupDrawInterfaceLayers>b__1585_24() in tModLoader\Terraria\Main.cs:line 34513
   at Terraria.UI.LegacyGameInterfaceLayer.DrawSelf() in tModLoader\Terraria\UI\LegacyGameInterfaceLayer.cs:line 12
   at Terraria.UI.GameInterfaceLayer.Draw() in tModLoader\Terraria\UI\GameInterfaceLayer.cs:line 41
   at Terraria.Main.DrawInterface(GameTime gameTime) in tModLoader\Terraria\Main.cs:line 34577
   at Terraria.Main.DoDraw(GameTime gameTime) in tModLoader\Terraria\Main.cs:line 49442
   at Terraria.Main.Draw(GameTime gameTime) in tModLoader\Terraria\Main.cs:line 48755
   at Microsoft.Xna.Framework.Game.Tick() in D:\a\tModLoader\tModLoader\FNA\src\Game.cs:line 562
   at Microsoft.Xna.Framework.Game.RunLoop() in D:\a\tModLoader\tModLoader\FNA\src\Game.cs:line 878
   at Microsoft.Xna.Framework.Game.Run() in D:\a\tModLoader\tModLoader\FNA\src\Game.cs:line 419
   at Terraria.Program.LaunchGame_(Boolean isServer) in tModLoader\Terraria\Program.cs:line 231
   at Terraria.Program.LaunchGame(String[] args, Boolean monoArgs) in tModLoader\Terraria\Program.cs:line 192
   at MonoLaunch.<>c__DisplayClass1_0.<Main>b__0() in tModLoader\Terraria\MonoLaunch.cs:line 56
   at System.Threading.Thread.StartCallback()
```

### Steps to reproduce

1. Make a player and join master mode world, put an accessory into every slot, save&quit.
2. Restart the game. (important!)
3. Open the game, select that player and join a non-master mode world.

### Expected Behavior

UI is visible

### Actual Behavior

UI is invisible

### Reproduction frequency

100%
The issue disappears if you have entered and left a world atleast once within the same game session. However, the "darkened" UI effect still occurs.

### Additional Information

Happens on stable and preview, not tested 1.4.4
This issue may also appear with the slot given by Demon Heart, not tested

**Comment by JavidPack** [comment]
Wasn't able to replicate on 1.4.4. (I misread which version this was for.)

**Comment by direwolf420** [comment]
Just tried to replicate it again, the UI disappearance doesn't happen. However, `Additionally, switching to the "Equipment" view makes the UI appear again as long as that view is active, except now it has the same "darkened" appearance as the inactive master mode slot.` still does happen.

**Comment by Chicken-Bones** [comment]
This should have been fixed by #3310 can you please test @direwolf420

**Comment by direwolf420** [comment]
not able to reproduce. the slot coloring with master mode slot still occurs if you switch to the "Misc Equips" page though
